src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -16,18 +16,217 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
-    internal sealed class SourceOrdinaryMethodSymbol : SourceOrdinaryMethodSymbolBase
+    internal class SourceOrdinaryMethodSymbol : SourceOrdinaryMethodSymbolBase
     {
-        private readonly TypeSymbol _explicitInterfaceType;
-
-        private readonly TypeParameterInfo _typeParameterInfo;
-
         /// <summary>
-        /// If this symbol represents a partial method definition or implementation part, its other part (if any).
-        /// This should be set, if at all, before this symbol appears among the members of its owner.  
-        /// The implementation part is not listed among the "members" of the enclosing type.
+        /// Specialized subclass of SourceOrdinaryMethodSymbol for less common cases.  Specifically, we only use this
+        /// for methods that are:
+        /// <list type="number">
+        /// <item>Generic.</item>
+        /// <item>An explicit interface implementation.</item>
+        /// <item>Partial.</item>
+        /// </list>
         /// </summary>
-        private SourceOrdinaryMethodSymbol _otherPartOfPartial;
+        private sealed class SourceOrdinaryMethodSymbolUncommon : SourceOrdinaryMethodSymbol
+        {
+            private readonly TypeSymbol _explicitInterfaceType;
+
+            private readonly TypeParameterInfo _typeParameterInfo;
+
+            /// <summary>
+            /// If this symbol represents a partial method definition or implementation part, its other part (if any).
+            /// This should be set, if at all, before this symbol appears among the members of its owner.  
+            /// The implementation part is not listed among the "members" of the enclosing type.
+            /// </summary>
+            private SourceOrdinaryMethodSymbol _otherPartOfPartial;
+
+            public SourceOrdinaryMethodSymbolUncommon(
+                NamedTypeSymbol containingType,
+                TypeSymbol explicitInterfaceType,
+                string name,
+                Location location,
+                MethodDeclarationSyntax syntax,
+                MethodKind methodKind,
+                bool isNullableAnalysisEnabled,
+                BindingDiagnosticBag diagnostics)
+                : base(containingType, name, location, syntax, methodKind, isNullableAnalysisEnabled, diagnostics)
+            {
+                _explicitInterfaceType = explicitInterfaceType;
+
+                _typeParameterInfo = new TypeParameterInfo { LazyTypeParameters = MakeTypeParameters(syntax, diagnostics) };
+            }
+
+            protected sealed override TypeSymbol ExplicitInterfaceType => _explicitInterfaceType;
+            internal override SourceOrdinaryMethodSymbol OtherPartOfPartial => _otherPartOfPartial;
+
```

Comment:
trying to figure out if there's a good way to manage code-moving in this PR.  These methods moved from SourceOrdinaryMethodSymbol into SourceOrdinaryMethodSymbolUncommon.  Unfortunately, github doesn't show moves well.  So hard to ascertain if it's a pure move.